### DISCUSSION TITLE
Github settings via file (e.g. .github/settings)

### DISCUSSION THREAD

**Discussion by fmenezes** [community/community#20027] [Other Feature Feedback, Questions, & Ideas] [answered]
Today settings are managed either [via UI](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features) and some via HTTP REST APIs (e.g. [branch protection](https://docs.github.com/en/rest/branches/branch-protection), [actions](https://docs.github.com/en/rest/actions) and others).

I think for some of us it would make sense to control some of these settings via file (my suggestion is `.github/settings` or something per feature like `.github/settings/branch-protection`).

Especially for `branch-protection`, `actions` and `webhooks` which are traditionally CI/CD rules, controlling them via file would greatly improve visibility and collaboration (and GitOps).

**Comment by laughedelic** [comment, marked as answer]
I think these threads are related: 

- https://github.com/github-community/community/discussions/16007
- https://github.com/github-community/community/discussions/13991

And there are a couple of tools for this:
- https://github.com/probot/settings: **settings config at the repo level, as you are suggesting**
- https://github.com/github/safe-settings: centralized org settings

I agree that it would be nice to have native official support for this functionality.
